### PR TITLE
feat: adds centralized wsman error struct and decode for boot and kvm

### PR DESCRIPTION
**Pull request by matt-primrose**

### CODE REVIEW

File: pkg/wsman/amt/boot/decoder.go [hunk 1]
```diff
@@ -55,3 +60,17 @@ func (i IDERBootDevice) String() string {
 
 	return ValueNotFound
 }
+
+func DecodeWSMANError(subCode string) error {
+	if len(subCode) > 0 {
+		modifiedString := strings.Split(subCode, ":")[1]
+		switch modifiedString {
+		case "AccessDenied":
+			return errors.New("User Consent is required for this action")
```
Comment (rsdmike) [inline pkg/wsman/amt/boot/decoder.go:69]: I was thinking this remains an access denied error and the user consent part is inferred in Console. Reason being, if we ever consider AMT users beyond an admin user, then its possible it is a permissions issue.

File: pkg/wsman/amt/boot/decoder.go [hunk 2]
```diff
@@ -55,3 +60,17 @@ func (i IDERBootDevice) String() string {
 
 	return ValueNotFound
 }
+
+func DecodeWSMANError(subCode string) error {
```
Comment (rsdmike) [inline pkg/wsman/amt/boot/decoder.go:64]: Is this just for this package? do we want to centralize an error handling function? or are they specific enough to warrant a function per package.

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: pkg/wsman/amt/boot/decoder.go (hunk 1 above)
Comment (matt-primrose) [reply pkg/wsman/amt/boot/decoder.go]: Implemented a centralized AMT Error to return this in a Go way

File: pkg/wsman/amt/boot/decoder.go (hunk 2 above)
Comment (matt-primrose) [reply pkg/wsman/amt/boot/decoder.go]: centralized!

**Review by rsdmike — APPROVED** [review]

**Pull request merged**